### ISSUE TITLE
Unstyled lists are not recognized by iOS/Voiceover

### ISSUE THREAD

**Issue by gnomeontherun** [vmware/clarity#3563]
## Describe the bug

Unstyled lists are not recognized by iOS/VoiceOver. The Unstyled List example is implemented with list-style-type:none. list-style-type:none can make it difficult for screen reader users to differentiate list items from one another by preventing the announcement of the word "bullet" before each list item. 

## How to reproduce

https://clarity.design/documentation/lists

- Using an iPhone or iPad with Safari and VoiceOver, read the list in the Unstyled Lists example.
- Observe that the screen reader does not announce to the user that they are reading a list.
- Observe that the word "bullet" is NOT announced by the screen reader before each list item is announced

## Expected behavior

Use overflow:hidden to hide the list bullets as seen at http://chrislane.info/examples/listWithHiddenBullets.html

## Versions

**App**

* Angular: [e.g. 6]
* Node: [e.g. 8.10.0]
* Clarity: [e.g. 0.12.5]

**Device:**

* Type: [e.g. MacBook]
* OS: [e.g. iOS]
* Browser [e.g. chrome, safari]
* Version [e.g. 22]

## Additional notes

Update the documentation to warn Clarity consumers to avoid implementing complex list items, since Mac OS X/Safari/VoiceOver does not announce the bullets for complex list items even when "list-style: none" is not in use.

**Comment by sheribyrne** [comment]
This is definitely an issue (not just merely a BP), but I've made it low, because while confusing, it doesn't mislead significantly or prevent navigation as some of the other tickets do
